src/main/java/com/example/librarymanagement/service/impl/JournalServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.example.librarymanagement.service.impl;
+
+import com.example.librarymanagement.exception.EntityNotFoundException;
+import com.example.librarymanagement.model.entity.Journal;
+import com.example.librarymanagement.repository.JournalRepository;
+import com.example.librarymanagement.service.JournalService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class JournalServiceImpl implements JournalService {
+    private final JournalRepository journalRepository;
+
+    public JournalServiceImpl(JournalRepository journalRepository) {
+        this.journalRepository = journalRepository;
+    }
+
+    @Override
+    public Journal createJournal(Journal journal) {
+        return journalRepository.save(journal);
+    }
+
+    @Override
+    public Journal getJournalById(Long id) {
+        return journalRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Journal"));
+    }
+
+    @Override
+    public List<Journal> getJournalByBookPresenceId(Long bookPresenceId) {
+        return journalRepository.findByBookPresenceId(bookPresenceId);
+    }
+
+    @Override
+    public Journal getJournalByBookPresenceIdAndUserId(Long bookPresenceId, Long userId) {
```

Comment:
I could use the same book multiple times, so I can have several journal records for one user and bookPresenceId.